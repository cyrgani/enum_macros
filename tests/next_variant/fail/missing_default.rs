use enum_macros::NextVariant;

#[derive(NextVariant)]
enum Complex<T> {
    A { val: T },
    B(u8),
}

fn main() {}
