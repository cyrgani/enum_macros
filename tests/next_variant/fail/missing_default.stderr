error[E0277]: the trait bound `T: Default` is not satisfied
 --> tests/next_variant/fail/missing_default.rs:3:10
  |
3 | #[derive(NextVariant)]
  |          ^^^^^^^^^^^ the trait `Default` is not implemented for `T`
  |
  = note: this error originates in the derive macro `NextVariant` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
  |
4 | enum Complex<T: std::default::Default> {
  |               +++++++++++++++++++++++
